src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala

```diff
@@ -50,15 +51,15 @@ case class DepositIngestTask(deposit: Deposit, dansBagValidator: DansBagValidato
     trace(())
     logger.info(s"Ingesting $deposit into Dataverse")
 
-    for {
+    val result = for {
       validationResult <- dansBagValidator.validateBag(bagDirPath)
       _ <- Try {
-        if (!validationResult.isCompliant) throw RejectedDepositException(deposit,
+        if (!validationResult.isCompliant) throw RejectedDepositException(deposit.dir,
```

Comment:
The line above can become `_ = {`, once you are in a try-context you don't need to make it explicit.